From 3d5495c5d1174b1438554c6d9d73e40f0e6a57eb Mon Sep 17 00:00:00 2001
From: Vivek Goyal <vgoyal@redhat.com>
Date: Fri, 5 Sep 2014 18:56:09 -0400
Subject: [kernel] kexec: remove CONFIG_KEXEC dependency on crypto

Message-id: <1409943373-12669-35-git-send-email-vgoyal@redhat.com>
Patchwork-id: 91543
O-Subject: [RHEL7.1 PATCH 34/38] kexec: remove CONFIG_KEXEC dependency on crypto
Bugzilla: 852066
RH-Acked-by: Dave Anderson <anderson@redhat.com>
RH-Acked-by: Baoquan He <bhe@redhat.com>
RH-Acked-by: Dave Young <dyoung@redhat.com>
RH-Acked-by: Chao WANG <chaowang@redhat.com>

BZ:852066

commit b41d34b46aa91e0c83300c38eebc0b8762705e82
Author: Vivek Goyal <vgoyal@redhat.com>
Date: Fri, 29 Aug 2014 15:18:49 -0700

New system call depends on crypto.  As it did not have a separate config
option, CONFIG_KEXEC was modified to select CRYPTO and CRYPTO_SHA256.

But now previous patch introduced a new config option for new syscall.
So CONFIG_KEXEC does not require crypto.  Remove that dependency.

Signed-off-by: Vivek Goyal <vgoyal@redhat.com>
Cc: Eric Biederman <ebiederm@xmission.com>
Cc: H. Peter Anvin <hpa@zytor.com>
Cc: Shaun Ruffell <sruffell@digium.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/arm/Kconfig b/arch/arm/Kconfig
index a77cc13..d1387fc 100644
--- a/arch/arm/Kconfig
+++ b/arch/arm/Kconfig
@@ -2020,8 +2020,6 @@ config XIP_PHYS_ADDR
 config KEXEC
  bool "Kexec system call (EXPERIMENTAL)"
  depends on (!SMP || PM_SLEEP_SMP)
- select CRYPTO
- select CRYPTO_SHA256
  help
    kexec is a system call that implements the ability to shutdown your
    current kernel, and to start another kernel.  It is like a reboot
diff --git a/arch/ia64/Kconfig b/arch/ia64/Kconfig
index e563bc0..1a2b774 100644
--- a/arch/ia64/Kconfig
+++ b/arch/ia64/Kconfig
@@ -556,8 +556,6 @@ source "drivers/sn/Kconfig"
 config KEXEC
  bool "kexec system call"
  depends on !IA64_HP_SIM && (!SMP || HOTPLUG_CPU)
- select CRYPTO
- select CRYPTO_SHA256
  help
    kexec is a system call that implements the ability to shutdown your
    current kernel, and to start another kernel.  It is like a reboot
diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
index 882ef36..e53e2b4 100644
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -2328,8 +2328,6 @@ source "kernel/Kconfig.preempt"
 
 config KEXEC
  bool "Kexec system call"
- select CRYPTO
- select CRYPTO_SHA256
  help
    kexec is a system call that implements the ability to shutdown your
    current kernel, and to start another kernel.  It is like a reboot
diff --git a/arch/powerpc/Kconfig b/arch/powerpc/Kconfig
index 877c786..97a7e48 100644
--- a/arch/powerpc/Kconfig
+++ b/arch/powerpc/Kconfig
@@ -387,8 +387,6 @@ config PPC64_SUPPORTS_MEMORY_FAILURE
 config KEXEC
  bool "kexec system call"
  depends on (PPC_BOOK3S || FSL_BOOKE || (44x && !SMP))
- select CRYPTO
- select CRYPTO_SHA256
  help
    kexec is a system call that implements the ability to shutdown your
    current kernel, and to start another kernel.  It is like a reboot
diff --git a/arch/s390/Kconfig b/arch/s390/Kconfig
index 2d56711..f800b80 100644
--- a/arch/s390/Kconfig
+++ b/arch/s390/Kconfig
@@ -48,8 +48,6 @@ config ARCH_SUPPORTS_DEBUG_PAGEALLOC
 
 config KEXEC
  def_bool y
- select CRYPTO
- select CRYPTO_SHA256
 
 config AUDIT_ARCH
  def_bool y
diff --git a/arch/sh/Kconfig b/arch/sh/Kconfig
index 3717f57..ff90bc4 100644
--- a/arch/sh/Kconfig
+++ b/arch/sh/Kconfig
@@ -634,8 +634,6 @@ source kernel/Kconfig.hz
 config KEXEC
  bool "kexec system call (EXPERIMENTAL)"
  depends on SUPERH32 && MMU
- select CRYPTO
- select CRYPTO_SHA256
  help
    kexec is a system call that implements the ability to shutdown your
    current kernel, and to start another kernel.  It is like a reboot
diff --git a/arch/tile/Kconfig b/arch/tile/Kconfig
index 6822fdf..3aa3766 100644
--- a/arch/tile/Kconfig
+++ b/arch/tile/Kconfig
@@ -172,8 +172,6 @@ source "kernel/Kconfig.hz"
 
 config KEXEC
  bool "kexec system call"
- select CRYPTO
- select CRYPTO_SHA256
  ---help---
    kexec is a system call that implements the ability to shutdown your
    current kernel, and to start another kernel.  It is like a reboot
-- 
1.7.1